arduino.go

```diff
@@ -0,0 +1,156 @@
+// Package arduino helps to talk to
+// an Arduino connected via USB (to a Raspberry Pi).
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+
+	"context"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"strings"
+	"sync"
+)
+
+var (
+	// AppFs is an abstraction of the file system
+	// to allow mocking in tests.
+	AppFs = afero.NewOsFs()
+)
+
+const (
+	// ReceiveCmd is the command that needs to b e sent to the Arduino to start
+	// receiving on Pin 0.
+	ReceiveCmd = "RF receive 0"
+	// ReceivePrefix is the prefix of raw signals read from the device file of the Arduino.
+	ReceivePrefix = "RF receive "
+
+	// ResetCmd is resetting the homeduino device to set it in an expected state
+	ResetCmd = "RESET"
+
+	// ResetCmdResponse is the response to ResetCmd homeduiono
+	ResetCmdResponse = "ready"
+)
+
+// ProcessorFunc are implementations for consuming line by line of the device
+type ProcessorFunc func(s string) bool
+
+// Device represents the device file of an Arduino
+// connected to the USB port.
+type Device struct {
+	afero.File
+	sync.Mutex
+	open bool
+}
+
+// OpenDevice opens the named device file for reading.
+func OpenDevice(name string) (*Device, error) {
```

Comment:
Do you mean the `bool` for open of the Device and the mutex? I thought that check is for not accessing a closed file for some reason :smile:

Reply:
I copied the basic device open, read, etc. frame from another exporter for a CO2 monitor (https://github.com/larsp/co2monitor/blob/master/meter/meter.go), also not sure weather it's needed TBH.

I just meant, there was also a lock on accessing the empty struct in the open function itself, which is not needed anymore, because the Device struct now is created by the open function (not a very important comment I made).